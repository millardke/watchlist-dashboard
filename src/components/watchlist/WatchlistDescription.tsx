import React from 'react';
import { shallowEqual, useSelector } from 'react-redux';
import styled from 'styled-components';
import { Button, Container } from '@material-ui/core';
import DeleteIcon from '@material-ui/icons/Delete';

import { RootStateInterface } from '../../model/RootStateModel';
import { SymbolDetailInterface } from '../../model/SymbolModel';

//
// -- Interface --

interface WatchlistDescriptionProps {
  code: string;
  handleClick: () => void;
}

//
// -- Styled Components --

const StyledButton = styled(Button)`
  width: 100%;
`;

const StyledContainer = styled(Container)`
  margin-bottom: 20px;
`;

const StyledSpan = styled.span`
  display: inline-block;
  margin-bottom: 10px;
`;

//
// -- WatchlistDescription Component --

const WatchlistDescription: React.FunctionComponent<WatchlistDescriptionProps> = (
  props: WatchlistDescriptionProps
) => {
  const details: SymbolDetailInterface = useSelector(
    (state: RootStateInterface) => state.watchlist.details,
    shallowEqual
  );

  return (
    <StyledContainer>
      <StyledSpan>Name: {details.name}</StyledSpan>
      <StyledSpan>Description: {details.description}</StyledSpan>
      <StyledButton
        variant="contained"
        color="secondary"
        startIcon={<DeleteIcon />}
        onClick={() => props.handleClick()}
      >
        Remove {props.code}
      </StyledButton>
    </StyledContainer>
  );
};

export default WatchlistDescription;
