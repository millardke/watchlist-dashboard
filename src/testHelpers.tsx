import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';

export const MockProvider = (children: any, providedStore: any) => {
  const middlewares: any = [];
  const mockStore = configureStore(middlewares);
  const store = mockStore(providedStore);
  const wrapper = mount(<Provider store={store}>{children}</Provider>);

  return wrapper;
};

export const MockStore = () => {
  return {
    watchlist: {
      details: {},
      list: [],
      selected: undefined,
      error: undefined,
    },
  };
};
