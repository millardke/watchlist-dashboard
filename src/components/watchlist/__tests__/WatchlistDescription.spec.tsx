import React from 'react';

import { MockProvider, MockStore } from '../../../testHelpers';
import WatchlistDescription from '../WatchlistDescription';

describe('WatchlistDescription Component', () => {
  it('should render WatchlistDescription', () => {
    const component = MockProvider(<WatchlistDescription />, MockStore);
    expect(component).toMatchSnapshot();
  });
});
