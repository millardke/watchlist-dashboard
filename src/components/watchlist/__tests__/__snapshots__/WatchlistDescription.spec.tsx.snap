// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WatchlistDescription Component should render WatchlistDescription 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <WatchlistDescription>
    <Styled(WithStyles(ForwardRef(Container)))>
      <WithStyles(ForwardRef(Container))
        className="sc-AxirZ fHEVYp"
      >
        <ForwardRef(Container)
          className="sc-AxirZ fHEVYp"
          classes={
            Object {
              "disableGutters": "MuiContainer-disableGutters",
              "fixed": "MuiContainer-fixed",
              "maxWidthLg": "MuiContainer-maxWidthLg",
              "maxWidthMd": "MuiContainer-maxWidthMd",
              "maxWidthSm": "MuiContainer-maxWidthSm",
              "maxWidthXl": "MuiContainer-maxWidthXl",
              "maxWidthXs": "MuiContainer-maxWidthXs",
              "root": "MuiContainer-root",
            }
          }
        >
          <div
            className="MuiContainer-root sc-AxirZ fHEVYp MuiContainer-maxWidthLg"
          >
            <styled.span>
              <span
                className="sc-AxiKw eCALDy"
              >
                Name: 
              </span>
            </styled.span>
            <styled.span>
              <span
                className="sc-AxiKw eCALDy"
              >
                Description: 
              </span>
            </styled.span>
            <Styled(WithStyles(ForwardRef(Button)))
              color="secondary"
              onClick={[Function]}
              startIcon={<UNDEFINED />}
              variant="contained"
            >
              <WithStyles(ForwardRef(Button))
                className="sc-AxjAm jdsAYC"
                color="secondary"
                onClick={[Function]}
                startIcon={<UNDEFINED />}
                variant="contained"
              >
                <ForwardRef(Button)
                  className="sc-AxjAm jdsAYC"
                  classes={
                    Object {
                      "colorInherit": "MuiButton-colorInherit",
                      "contained": "MuiButton-contained",
                      "containedPrimary": "MuiButton-containedPrimary",
                      "containedSecondary": "MuiButton-containedSecondary",
                      "containedSizeLarge": "MuiButton-containedSizeLarge",
                      "containedSizeSmall": "MuiButton-containedSizeSmall",
                      "disableElevation": "MuiButton-disableElevation",
                      "disabled": "Mui-disabled",
                      "endIcon": "MuiButton-endIcon",
                      "focusVisible": "Mui-focusVisible",
                      "fullWidth": "MuiButton-fullWidth",
                      "iconSizeLarge": "MuiButton-iconSizeLarge",
                      "iconSizeMedium": "MuiButton-iconSizeMedium",
                      "iconSizeSmall": "MuiButton-iconSizeSmall",
                      "label": "MuiButton-label",
                      "outlined": "MuiButton-outlined",
                      "outlinedPrimary": "MuiButton-outlinedPrimary",
                      "outlinedSecondary": "MuiButton-outlinedSecondary",
                      "outlinedSizeLarge": "MuiButton-outlinedSizeLarge",
                      "outlinedSizeSmall": "MuiButton-outlinedSizeSmall",
                      "root": "MuiButton-root",
                      "sizeLarge": "MuiButton-sizeLarge",
                      "sizeSmall": "MuiButton-sizeSmall",
                      "startIcon": "MuiButton-startIcon",
                      "text": "MuiButton-text",
                      "textPrimary": "MuiButton-textPrimary",
                      "textSecondary": "MuiButton-textSecondary",
                      "textSizeLarge": "MuiButton-textSizeLarge",
                      "textSizeSmall": "MuiButton-textSizeSmall",
                    }
                  }
                  color="secondary"
                  onClick={[Function]}
                  startIcon={<UNDEFINED />}
                  variant="contained"
                >
                  <WithStyles(ForwardRef(ButtonBase))
                    className="MuiButton-root MuiButton-contained sc-AxjAm jdsAYC MuiButton-containedSecondary"
                    component="button"
                    disabled={false}
                    focusRipple={true}
                    focusVisibleClassName="Mui-focusVisible"
                    onClick={[Function]}
                    type="button"
                  >
                    <ForwardRef(ButtonBase)
                      className="MuiButton-root MuiButton-contained sc-AxjAm jdsAYC MuiButton-containedSecondary"
                      classes={
                        Object {
                          "disabled": "Mui-disabled",
                          "focusVisible": "Mui-focusVisible",
                          "root": "MuiButtonBase-root",
                        }
                      }
                      component="button"
                      disabled={false}
                      focusRipple={true}
                      focusVisibleClassName="Mui-focusVisible"
                      onClick={[Function]}
                      type="button"
                    >
                      <button
                        className="MuiButtonBase-root MuiButton-root MuiButton-contained sc-AxjAm jdsAYC MuiButton-containedSecondary"
                        disabled={false}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onDragLeave={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        tabIndex={0}
                        type="button"
                      >
                        <span
                          className="MuiButton-label"
                        >
                          <span
                            className="MuiButton-startIcon MuiButton-iconSizeMedium"
                          >
                            <ForwardRef>
                              <WithStyles(ForwardRef(SvgIcon))>
                                <ForwardRef(SvgIcon)
                                  classes={
                                    Object {
                                      "colorAction": "MuiSvgIcon-colorAction",
                                      "colorDisabled": "MuiSvgIcon-colorDisabled",
                                      "colorError": "MuiSvgIcon-colorError",
                                      "colorPrimary": "MuiSvgIcon-colorPrimary",
                                      "colorSecondary": "MuiSvgIcon-colorSecondary",
                                      "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                      "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                      "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                      "root": "MuiSvgIcon-root",
                                    }
                                  }
                                >
                                  <svg
                                    aria-hidden={true}
                                    className="MuiSvgIcon-root"
                                    focusable="false"
                                    viewBox="0 0 24 24"
                                  >
                                    <path
                                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                                    />
                                  </svg>
                                </ForwardRef(SvgIcon)>
                              </WithStyles(ForwardRef(SvgIcon))>
                            </ForwardRef>
                          </span>
                          Remove 
                        </span>
                        <WithStyles(memo)
                          center={false}
                        >
                          <ForwardRef(TouchRipple)
                            center={false}
                            classes={
                              Object {
                                "child": "MuiTouchRipple-child",
                                "childLeaving": "MuiTouchRipple-childLeaving",
                                "childPulsate": "MuiTouchRipple-childPulsate",
                                "ripple": "MuiTouchRipple-ripple",
                                "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                                "rippleVisible": "MuiTouchRipple-rippleVisible",
                                "root": "MuiTouchRipple-root",
                              }
                            }
                          >
                            <span
                              className="MuiTouchRipple-root"
                            >
                              <TransitionGroup
                                childFactory={[Function]}
                                component={null}
                                exit={true}
                              />
                            </span>
                          </ForwardRef(TouchRipple)>
                        </WithStyles(memo)>
                      </button>
                    </ForwardRef(ButtonBase)>
                  </WithStyles(ForwardRef(ButtonBase))>
                </ForwardRef(Button)>
              </WithStyles(ForwardRef(Button))>
            </Styled(WithStyles(ForwardRef(Button)))>
          </div>
        </ForwardRef(Container)>
      </WithStyles(ForwardRef(Container))>
    </Styled(WithStyles(ForwardRef(Container)))>
  </WatchlistDescription>
</Provider>
`;
